src/EventGridEmulator.Tests/EmulatorValidationTests.cs

```diff
@@ -73,9 +73,42 @@ public async Task ValidatePublishAndSubscribeRoundTripForCloudEvent()
         // Assert that the message was successfully received
         var @event = JsonSerializer.Deserialize<JsonObject[]>(message) ?? throw new NullReferenceException("Message cannot be deserialized");
         var result = @event.Single()["data"].Deserialize<DataModel>();
-        var receivedTopic = @event.Single()["source"].Deserialize<string>();
+        var receivedSource = @event.Single()["source"].Deserialize<string>();
         Assert.Equal("data", result?.Some);
-        Assert.Equal($"{SubscriberConstants.DefaultTopicValue}{ExpectedTopic}", receivedTopic);
+        Assert.Equal($"{SubscriberConstants.DefaultTopicValue}{ExpectedTopic}", receivedSource);
+    }
+
+    [Fact]
+    public async Task ValidatePublishAndSubscribeRoundTripForCloudEventPreserveOriginalSource()
+    {
+        // Define the handler to intercept the message
+        var message = string.Empty;
+        var originalSource = "SOME_SOURCE_VALUE";
+        HttpMessageHandler handler = new TestHandler(requestAction => message = requestAction);
```

Comment:
I think this is Disposable. Maybe you can also update other locations

```suggestion
        using HttpMessageHandler handler = new TestHandler(requestAction => message = requestAction);
```

Reply:
done